libraries/AP_Hott_Telem/AP_Hott_Telem.cpp

```diff
@@ -0,0 +1,358 @@
+        uint8_t sensor_id = 0xA0;     //#04 constant (?) value 0xa0
+        uint16_t alarm; //#05
+        uint8_t flight_direction;     //#07 flight direction in 2 degreees/step (1 = 2degrees);
+        uint16_t gps_speed_kmh;       //#08 km/h
+        uint8_t pos_NS;               //#10 north = 0, south = 1
+        uint16_t pos_NS_dm;           //#11 degree minutes
+        uint16_t pos_NS_sec;          //#13 position seconds
+        uint8_t pos_EW;               //#15 east = 0, west = 1
+        uint16_t pos_EW_dm;           //#16 degree minutes
+        uint16_t pos_EW_sec;          //#18 position seconds
+        uint16_t home_distance;       //#20 meters
+        uint16_t altitude;            //#22 meters. Value of 500 = 0m
+        uint16_t climbrate;           //#24 m/s 0.01m/s resolution. Value of 30000 = 0.00 m/s
+        uint8_t climbrate3s;          //#26 climbrate in m/3s resolution, value of 120 = 0 m/3s
+        uint8_t gps_satelites;        //#27 sat count
+        uint8_t gps_fix_char;         //#28 GPS fix character. display, 'D' = DGPS, '2' = 2D, '3' = 3D, '-' = no fix
+        uint8_t home_direction;       //#29 direction from starting point to Model position (2 degree steps)
+        int16_t vel_north;            //#30 velocity north mm/s
+        uint8_t speed_acc;            //#32 speed accuracy cm/s
+        uint8_t gps_time_h;           //#33 UTC time hours
+        uint8_t gps_time_m;           //#34 UTC time minutes
+        uint8_t gps_time_s;           //#35 UTC time seconds
+        uint8_t gps_time_hs;          //#36 UTC time 0.01s units
+        int16_t vel_east;             //#37 velocity east mm/s
+        uint8_t horiz_acc;            //#39 horizontal accuracy dm
+        uint16_t msl_altitude;        //#40 mean sea level altitude
+        uint8_t gps_fix_char2;        //#42 GPS fix character2??. display, 'D' = DGPS, '2' = 2D, '3' = 3D, '-' = no fix
+        uint8_t version;       //#43  0: GPS Graupner #33600
+        uint8_t status1;
+        uint8_t status2;
+        uint8_t stop_byte = 0x7d;
+    } msg {};
+
+    AP_GPS &gps = AP::gps();
+    Location loc;
+
+    {
+        WITH_SEMAPHORE(gps.get_semaphore());
+        loc = gps.location();
+        msg.flight_direction = uint16_t(gps.ground_course() * 0.5 + 0.5);
+        msg.gps_speed_kmh = uint16_t(gps.ground_speed() * 3.6 + 0.5);
+        float sacc, hacc;
+        if (gps.speed_accuracy(sacc)) {
+            msg.speed_acc = sacc * 100 + 0.5;
+        }
+        if (gps.horizontal_accuracy(hacc)) {
+            msg.horiz_acc = hacc * 100 + 0.5;
+        }
+    }
+
+    float lat = loc.lat * 1.0e-7;
+    float lon = loc.lng * 1.0e-7;
+
+    uint16_t dm, sec;
+    GPS_to_DMS(lat, msg.pos_NS, dm, sec);
+    msg.pos_NS_dm = dm;
+    msg.pos_NS_sec = sec;
+
+    GPS_to_DMS(lon, msg.pos_EW, dm, sec);
+    msg.pos_EW_dm = dm;
+    msg.pos_EW_sec = sec;
+
+    AP_AHRS &ahrs = AP::ahrs();
+    Vector2f home_vec;
+    float posD = 0;
```

Comment:
Slight oddity here in that we set other message fields while still holding the semaphore - why not this one?  Or not set the message fields generally while holding the semaphore.